internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -145,25 +145,32 @@ func resourceKustoCluster() *pluginsdk.Resource {
 			"virtual_network_configuration": {
 				Type:     pluginsdk.TypeList,
 				Optional: true,
-				ForceNew: true,
 				MaxItems: 1,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
 						"subnet_id": {
 							Type:         pluginsdk.TypeString,
 							Required:     true,
+							ForceNew:     true,
 							ValidateFunc: commonids.ValidateSubnetID,
 						},
 						"engine_public_ip_id": {
 							Type:         pluginsdk.TypeString,
 							Required:     true,
+							ForceNew:     true,
 							ValidateFunc: networkValidate.PublicIpAddressID,
 						},
 						"data_management_public_ip_id": {
 							Type:         pluginsdk.TypeString,
 							Required:     true,
+							ForceNew:     true,
 							ValidateFunc: networkValidate.PublicIpAddressID,
 						},
+						"enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
```

Comment:
We should add a validation check in the `create` and `update` function to verify that the `public_network_access_enabled` is also set to `false` if `enabled` is set to `false` since in the documentation my exposed a note stating that if `enabled` is set to `false` it would also set `public_network_access_enabled` to `false` as well. Without the validation check in the `create` and `update` functions this would cause a diff on the next `plan` or `apply`.

Reply:
Ok, I will add that validation check code block in the createUpdate function and rerun tests.